core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -101,20 +100,34 @@ private String fileClass() {
 
   protected ManifestReader(
       InputFile file,
+      int specId,
       Map<Integer, PartitionSpec> specsById,
       InheritableMetadata inheritableMetadata,
       FileType content) {
     this.file = file;
     this.inheritableMetadata = inheritableMetadata;
     this.content = content;
 
+    if (specsById != null) {
+      this.spec = specsById.get(specId);
+      Preconditions.checkArgument(
+          spec != null, "Could not find PartitionSpec for specId: %s", specId);
```

Comment:
In Iceberg, we don't use class or variable names in error messages. The people reading these messages are not Iceberg developers so it is better to just use normal words. Here, this should be `"Cannot find partition spec for id: %s"`

Reply:
Thanks for the advice